src/code/z_lifemeter.c

```diff
@@ -1,17 +1,437 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_lifemeter/LifeMeter_Init.s")
+static s16 sHeartsPrimColors[3][3] = { { 255, 70, 50 }, { 255, 190, 0 }, { 100, 100, 255 } };
+static s16 sHeartsEnvColors[3][3] = { { 50, 40, 60 }, { 255, 0, 0}, { 0, 0, 255 } };
+static s16 sHeartsPrimFactors[3][3] = { { 0, 0, 0 }, { 0, 120, -50}, { -155, 30, 205 } };
+static s16 sHeartsEnvFactors[3][3] = { { 0, 0, 0 }, { 205, -40, -60 }, { -50, -40, 195 } };
+static s16 sHeartsDDPrimColors[3][3] = { { 255, 255, 255 }, { 255, 190, 0 }, { 100, 100, 255 } };
+static s16 sHeartsDDEnvColors[3][3] = { { 200, 0,  0}, { 255, 0, 0 }, { 0, 0, 255 } };
+static s16 sHeartsDDPrimFactors[3][3] = { { 0, 0, 0 }, { 0, -65, -255 },{ -155, -155, 0 } };
+static s16 sHeartsDDEnvFactors[3][3] = { { 0, 0, 0 }, { 55, 0, 0 }, { -200, 0, 255 } };
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_lifemeter/LifeMeter_UpdateColors.s")
+s16 sBeatingHeartsDDPrim[3];
+s16 sBeatingHeartsDDEnv[3];
+s16 sHeartsDDPrim[2][3];
+s16 sHeartsDDEnv[2][3];
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_lifemeter/func_80100A80.s")
+extern TexturePtr D_02000000; // Empty heart texture
+extern TexturePtr D_02000400; // Full heart texture
+extern TexturePtr D_02000500; // Empty Double Defense Heart texture
+extern TexturePtr D_02000900; // Full Double Defense Heart texture
+extern TexturePtr heartQuarterTexture; 
+extern TexturePtr heartHalfTexture; 
+extern TexturePtr heartThreeQuarterTexture; 
+extern TexturePtr ddHeartQuarterTexture; 
+extern TexturePtr ddHeartHalfTexture; 
+extern TexturePtr ddHeartThreeQuarterTexture; 
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_lifemeter/func_80100AA0.s")
+TexturePtr HeartTextures[] = {
+    &D_02000400, // Full
+    &heartQuarterTexture, &heartQuarterTexture, &heartQuarterTexture, &heartQuarterTexture, &heartQuarterTexture,
+    &heartHalfTexture, &heartHalfTexture, &heartHalfTexture, &heartHalfTexture, &heartHalfTexture,
+    &heartThreeQuarterTexture, &heartThreeQuarterTexture, &heartThreeQuarterTexture, &heartThreeQuarterTexture, &heartThreeQuarterTexture
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_lifemeter/func_80100AF0.s")
+TexturePtr HeartDDTextures[] = {
+    &D_02000900, // Full
+    &ddHeartQuarterTexture, &ddHeartQuarterTexture, &ddHeartQuarterTexture, &ddHeartQuarterTexture, &ddHeartQuarterTexture, 
+    &ddHeartHalfTexture, &ddHeartHalfTexture, &ddHeartHalfTexture, &ddHeartHalfTexture, &ddHeartHalfTexture,
+    &ddHeartThreeQuarterTexture, &ddHeartThreeQuarterTexture, &ddHeartThreeQuarterTexture, &ddHeartThreeQuarterTexture, &ddHeartThreeQuarterTexture
+};
+void LifeMeter_Init(GlobalContext* globalCtx) {
```

Comment:
Leave an empty line between data and functions.
```suggestion
};

void LifeMeter_Init(GlobalContext* globalCtx) {
```